cmd/minikube/cmd/config/open.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/util"
+)
+
+var (
+	namespace         string
+	https             bool
+	addonsURLMode     bool
+	addonsURLFormat   string
+	addonsURLTemplate *template.Template
+)
+
+const defaultAddonsFormatTemplate = "http://{{.IP}}:{{.Port}}"
+
+var addonsOpenCmd = &cobra.Command{
+	Use:   "open ADDON_NAME",
+	Short: "Opens the addon w/ADDON_NAME within minikube (example: minikube addons open dashboard). For a list of available addons use: minikube addons list ",
+	Long:  "Opens the addon w/ADDON_NAME within minikube (example: minikube addons open dashboard). For a list of available addons use: minikube addons list ",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		t, err := template.New("addonsURL").Parse(addonsURLFormat)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "The value passed to --format is invalid:\n\n", err)
+			os.Exit(1)
+		}
+		addonsURLTemplate = t
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			fmt.Fprintln(os.Stderr, "usage: minikube addons open ADDON_NAME")
+			os.Exit(1)
+		}
+		addonName := args[0]
+		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+		defer api.Close()
+
+		cluster.EnsureMinikubeRunningOrExit(api, 1)
+		if _, ok := assets.Addons[addonName]; !ok { // validate addon input
+			fmt.Fprintln(os.Stderr, fmt.Sprintf("addon '%s' is not a valid addon packaged with minikube", addonName))
```

Comment:
Is there a way for the user to list the valid add-ons? Could you add that to this error?